### DISCUSSION TITLE
Debug Labels for atom Families

### DISCUSSION THREAD

**Discussion by stevemolitor** [pmndrs/jotai#2254] [General]
Is it possible to extend `jotai/babel` to support adding debug labels for `atomFamily`? Ideally, the debug label would include the stringified atom family param, but I can see how that could get tricky.

**Comment by dai-shi** [comment]
> but I can see how that could get tricky.

yeah.
but, anyone can feel free to investigate and experiment the feasibility.

**Comment by chriszrc** [comment]
At least we can do it manually for now, but it would be nice if babel did it automatically:

```
export const layerAccordionsFamily = atomFamily((param: string) => {
  const atomMember = atomWithStorage(
    `layerAccordions.${param}`,
    false
  );
  atomMember.debugLabel = `layerAccordionsFamily.${param}`;
  return atomMember;
}, deepEqual);
```

Also, is there any way to add debugLabels when we use `splitAtom`?

**Reply by dai-shi** [reply]
It would be technically possible for atomFamily for some cases, but would not work for all cases.
For splitAtom, it would be difficult. What you could do is to attach the debug label manually with subscribing to it.
